### ISSUE TITLE
adding new values to a lazyObservable array

### ISSUE THREAD

**Issue by peebles** [mobxjs/mobx-utils#79]
My lazyObservable array works great the first time the react component is mounted, and later if I do something like

    list.current()[ 5 ].name = "jones"

The table showing the list is re-rendered and I see the new name.  But if I

    list.current().push( newPerson )

My table is not growing by one.

```javascript
    { list.current().map( person => 
      <Row key={person.id} ...
          <Cell>{person.name}</Cell>
```

What am I missing?

**Comment by peebles** [comment]
I probably should note that the list.current().push() is happening after an async fetch of data from a backend.  Interestingly enough, if I push onto the list prior to making the async call, I do see the list grow.  This mobx newbee must be missing something....

**Comment by mattruby** [comment]
If possible, please create a small reproduction of your issue on
codesandbox.io or jsfiddle.

**Comment by peebles** [comment]
I'll try.  I have been trying other things ... like @action.bound, but 
still haven't any success.  I had high hopes for @action.bound!

Is there any sort of minimal template I can start with for creating a 
fiddle with mobx/react and decorators?

**Comment by mattruby** [comment]
Lots of min templates. Check out codesandbox.io

**Comment by peebles** [comment]
Ok: https://codesandbox.io/s/53l4vqnm7x

When it runs, it will show a list of two names.  Click on add patient, 
and it tries to add a third.  You can see that the list does not 
re-render.  If you just type a space in index.js to cause it to 
re-render, you'll see three names.

So this appears to be representative of the problem I am seeing ... how 
to update a lazyObservable in the future, async.

**Comment by mattruby** [comment]
Soooo, I don't use `lazyObservable`.  But from what I can tell, I think you're using it wrong.  Here's a working example of it in action using fetch: https://codesandbox.io/s/zxq8prpwo3 Look at the console

I'm pretty sure you should not be messing with `lazyObservable.current()`'s results.  If you want to update the results, I think you should be calling '.refresh()' on the `lazyObservable`

**Comment by mattruby** [comment]
@peebles Here's your example (tweaked around a bit) working with `lazyObservable` https://codesandbox.io/s/jn8vqk4vz9

**Comment by peebles** [comment]
Yeah, I see.  I had also tried that, but I don't really want to go back 
to the server for a complete reload of the list.  My real addPatient() 
will hit the server to add a patient and the response is the added 
patient along with its new database id.  If I had to refetch the list, 
and the list was significantly large ...

I mean at that point, I'd just use a regular observable and load it in 
componetDidMount().

I am wondering now, if the "problem" is that, the lazyObservable is 
putting the initialValue in a box().  current() gets the value of the 
box, which itself is not the observable.  Seems perhaps only a box.set() 
will notify subscribers.

I may try my own hand at a lazyObservable that does not use a box().
